Python app has dependency issue which causes exception to be raised

Repro:

```bash
cd sample-apps/python
docker build -t sample-python-app-repro .
docker run --rm sample-python-app-repro
[2023-10-06 19:52:03 +0000] [1] [INFO] Starting gunicorn 20.1.0
[2023-10-06 19:52:03 +0000] [1] [INFO] Listening at: http://0.0.0.0:8080 (1)
[2023-10-06 19:52:03 +0000] [1] [INFO] Using worker: sync
[2023-10-06 19:52:03 +0000] [7] [INFO] Booting worker with pid: 7
[2023-10-06 19:52:03 +0000] [7] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/gunicorn/arbiter.py", line 589, in spawn_worker
    worker.init_process()
  File "/usr/local/lib/python3.10/site-packages/gunicorn/workers/base.py", line 134, in init_process
    self.load_wsgi()
  File "/usr/local/lib/python3.10/site-packages/gunicorn/workers/base.py", line 146, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/usr/local/lib/python3.10/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/local/lib/python3.10/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
    return self.load_wsgiapp()
  File "/usr/local/lib/python3.10/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/local/lib/python3.10/site-packages/gunicorn/util.py", line 359, in import_app
    mod = importlib.import_module(module)
  File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/usr/src/app/server.py", line 16, in <module>
    from flask import Flask
  File "/usr/local/lib/python3.10/site-packages/flask/__init__.py", line 5, in <module>
    from .app import Flask as Flask
  File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 30, in <module>
    from werkzeug.urls import url_quote
ImportError: cannot import name 'url_quote' from 'werkzeug.urls' (/usr/local/lib/python3.10/site-packages/werkzeug/urls.py)
[2023-10-06 19:52:03 +0000] [7] [INFO] Worker exiting (pid: 7)
[2023-10-06 19:52:03 +0000] [1] [INFO] Shutting down: Master
[2023-10-06 19:52:03 +0000] [1] [INFO] Reason: Worker failed to boot.
```

--- a/sample-apps/python/requirements.txt
+++ b/sample-apps/python/requirements.txt
@@ -1,3 +1,9 @@
-Flask==2.2.2
-gunicorn==20.1.0
-requests==2.28.1
+blinker==1.6.2
+click==8.1.7
+Flask==2.3.3
+gunicorn==21.2.0
+itsdangerous==2.1.2
+Jinja2==3.1.2
+MarkupSafe==2.1.3
+packaging==23.2
+Werkzeug==2.3.7

